src/Id/TableGenerator.php

```diff
@@ -118,7 +119,8 @@ public function nextValue($sequence)
             $platform = $this->conn->getDatabasePlatform();
             $sql      = 'SELECT sequence_value, sequence_increment_by'
                 . ' FROM ' . $platform->appendLockHint($this->generatorTableName, LockMode::PESSIMISTIC_WRITE)
-                . ' WHERE sequence_name = ? ' . $platform->getWriteLockSQL();
+                . ' WHERE sequence_name = ? '
+                . ($platform instanceof SQLServerPlatform ? '' : $platform->getWriteLockSQL());
```

Comment:
We should avoid this kind of `instanceof` checks. Why are we changing this deprecated class anyway?

Reply:
We changed the `SQLServerPlatform::getWriteLockSQL()` so that it actually spits out the right SQL, it was spitting out  an empty string.
Thus, after changing it, here for MSSQL it was adding both the lock hints after the `FROM` (correct for MSQL) and at the end (correct for all others).
So, the test was failing and we had to change it.
Since the class is deprecated anyway, I assumes it would be fine to hack it up. Otherwise we would have to make a deep refactoring to a deprecated class, so that we would use the query builder instead of directly SQL,

Reply:
> Thus, after changing it, here for MSSQL it was adding both the lock hints after the `FROM` (correct for MSQL) and at the end (correct for all others).

But that sounds correct to me, doesn't it?

> So, the test was failing and we had to change it.

Which test exactly?

We really cannot keep this, sorry. Not having platform switches all over the codebase is the reason why the platform classes exist.

Reply:
> > Thus, after changing it, here for MSSQL it was adding both the lock hints after the FROM (correct for MSQL) and at the end (correct for all others).
>
> But that sounds correct to me, doesn't it?

No, it should only add the extra SQL (the locking bit) either just after the `from` or at the end, and for MSSQL it was adding it in both locations.

> > So, the test was failing and we had to change it.
> 
> Which test exactly?
>
> We really cannot keep this, sorry. Not having platform switches all over the codebase is the reason why the platform classes exist.

I don't remember what test was breaking anymore.
Now I replaced building the raw SQL (which was there already) for building the SQL with the query builder, using the locking method I added.

I left my latest changes in a `fixup!` commit that should be squashed before merging.